src/base/QXmppOmemoIq.h

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2022 Melvin Keskin <melvo@olomono.de>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#ifndef QXMPPOMEMOIQ_H
+#define QXMPPOMEMOIQ_H
+
+#include "QXmppIq.h"
+#include "QXmppOmemoElement.h"
+
+#include <QSharedDataPointer>
+
+class QXmppOmemoIqPrivate;
+
+class QXMPP_EXPORT QXmppOmemoIq : public QXmppIq
+{
+public:
+    QXmppOmemoIq();
+    QXmppOmemoIq(const QXmppOmemoIq &);
+    ~QXmppOmemoIq() override;
+
+    QXmppOmemoIq &operator=(const QXmppOmemoIq &other);
+
+    std::optional<QXmppOmemoElement> omemoElement();
+    void setOmemoElement(const std::optional<QXmppOmemoElement> omemoElement);
```

Comment:
What's the idea of the optional here?
Also, you might want to inherit from OmemoElement, so you can directly access the attributes from the IQ and there's no need for an additional QSharedDataPointer.

Reply:
The OMEMO element does not need to be optional here. I changed that. The upcoming OMEMO manager creates an OMEMO element and sets it within the `QXmppOmemoIq`. Thus, I think it is easier if `QXmppOmemoIq` does not inherit from `QXmppOmemoElement`. Furthermore, the IQ stanza **is** not an OMEMO element, it **contains** one.

Reply:
> The upcoming OMEMO manager creates an OMEMO element and sets it within the QXmppOmemoIq. Thus, I think it is easier if QXmppOmemoIq does not inherit from QXmppOmemoElement.

Should be no big issue, you could e.g. add a `QXmppOmemoElement` constructor.

> Furthermore, the IQ stanza is not an OMEMO element, it contains one.

That's actually not the case, if you have a look at the XML. The IQ only consists of the omemo element, it's not like this: `<iq><omemo-iq><omemo-element/></omemo-iq></iq>`, but `<iq><omemo-element/></iq>` (and IQs have exactly one child element, so it wouldn't be possible to have any other extension).